transports/http/handlers/webhooks_handler.go

```diff
@@ -0,0 +1,93 @@
+package handler
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	webhook "github.com/libsv/bitcoin-hc/transports/http"
+)
+
+// nolint: godot
+// registerWebhook godoc.
+//
+//	@Summary Register new webhook
+//	@Tags webhooks
+//	@Accept json
+//	@Produce json
+//	@Success 200 {object} domains.Webhook
+//	@Router /webhook [post]
+//	@Param data body http.WebhookRequest true "Webhook to register"
+//
+// @Security Bearer
+func (h *Handler) registerWebhook(c *gin.Context) {
+	var reqBody webhook.WebhookRequest
+	err := c.Bind(&reqBody)
+
+	fmt.Println("WEBHOOK: ", reqBody)
+
+	if err == nil {
+
+		if reqBody.Url == "" {
+			c.JSON(http.StatusBadRequest, "Url is required")
+			return
+		}
+		var tHeader, token string
+
+		// If custom header is specified, use it, otherwise use default
+		if strings.ToLower(reqBody.RequiredAuth.Type) == "custom_header" {
+			tHeader = reqBody.RequiredAuth.Header
+			token = reqBody.RequiredAuth.Token
+		} else {
+			tHeader = "Authorization"
+			token = "Bearer " + reqBody.RequiredAuth.Token
+		}
+
+		webhook, err := h.services.Webhooks.GenerateWebhook(reqBody.Url, tHeader, token)
+		if err == nil {
+			c.JSON(http.StatusOK, webhook)
+		} else if webhook == nil {
+			c.JSON(http.StatusOK, err.Error())
+		}
+	}
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+	}
+}
+
+// nolint: godot
+// revokeWebhook godoc.
+//
+//	@Summary Revoke webhook
+//	@Tags webhooks
+//	@Accept */*
+//	@Success 200
+//	@Produce json
+//	@Router /webhook [delete]
+//	@Param url query string true "Url of webhook to revoke"
+//
+// @Security Bearer
+func (h *Handler) revokeWebhook(c *gin.Context) {
+	url := c.Query("url")
+	if url == "" {
+		c.JSON(http.StatusBadRequest, "Url param is required")
+		return
+	}
+	err := h.services.Webhooks.DeleteWebhook(url)
+
+	if err == nil {
+		c.JSON(http.StatusOK, "Webhook revoked")
+	} else {
+		c.JSON(http.StatusBadRequest, err.Error())
+	}
+}
+
+func (h *Handler) initRegisteredWehooksRoutes(router *gin.RouterGroup) {
+	webhooks := router.Group("")
+	{
+		webhooks.POST("/webhook", h.registerWebhook)
+		webhooks.DELETE("/webhook", h.revokeWebhook)
```

Comment:
IMHO it's good idea to have an endpoint to list a webhooks so you can check the status of it.